gprofiler/utils/node.py

```diff
@@ -0,0 +1,223 @@
+    def generate_perf_map():
+        debugger_url = _get_debugger_url()
+        sock = create_connection(debugger_url)
+        _load_dso(sock, module_path)
+
+    return generate_perf_map
+
+
+def _clean_up_wrapper(module_path: str, pid: int) -> Callable:
+    def clean_up():
+        debugger_url = _get_debugger_url()
+        sock = create_connection(debugger_url)
+        _stop_dso(sock, module_path)
+        shutil.rmtree(os.path.join("/tmp", "node_module"))
+        os.remove(os.path.join("/tmp", f"perf-{pid}.map"))
+
+    return clean_up
+
+
+def generate_map_for_node_processes():
+    """Iterates over all NodeJS processes, starts debugger for it, finds debugger URL,
+    copies node-linux-perf module into process' namespace, loads module and starts it.
+    After that it links it into gProfiler namespace's /tmp. This lets perf load it"""
+    processes = get_node_processes()
+    for process in processes:
+        try:
+            musl = is_musl(process)
+            node_version = get_exe_version(process)
+            node_major_version = node_version[1:].split(".")[0]
+            dest = _copy_module_into_process_ns(process, musl, node_major_version)
+            _start_debugger(process.pid)
+            pid_inside_ns = _get_pid_inside_ns(process.pid)
+            cp = run_in_ns(
+                ["pid", "mnt", "net"], _generate_perf_map_wrapper(dest), process.pid
+            )
+            if hasattr(cp, "stdout"):
+                logger.debug(cp.stdout.decode().strip())
+            if hasattr(cp, "stderr"):
+                logger.debug(cp.stderr.decode().strip())
+            if not is_same_ns(process, NsType.mnt.name):
+                _create_symlink_for_map(process, pid_inside_ns)
+        except Exception as e:
+            logger.warn(
+                f"Could not create debug symbols for pid {process.pid}. Reason: {e}"
+            )
+            logger.debug(traceback.format_exc())
+
+
+def clean_up_node_maps():
+    """Stops generating perf maps for each NodeJS process and cleans up generated maps"""
+    processes = get_node_processes()
+    for process in processes:
+        try:
+            node_version = get_exe_version(process)
+            node_major_version = node_version[1:].split(".")[0]
+            pid_inside_ns = _get_pid_inside_ns(process.pid)
+            libc = "musl" if is_musl(process) else "glibc"
+            dest = os.path.join("/tmp", "node_module", libc, node_major_version)
+            cp = run_in_ns(
+                ["pid", "mnt", "net"],
+                _clean_up_wrapper(dest, pid_inside_ns),
+                process.pid,
+            )
+            if hasattr(cp, "stdout"):
+                logger.debug(cp.stdout.decode().strip())
+            if hasattr(cp, "stderr"):
+                logger.debug(cp.stderr.decode().strip())
+            if not is_same_ns(process, NsType.mnt.name):
+                os.remove(os.path.join("/tmp", f"perf-{process.pid}.map"))
```

Comment:
Why? I don't think this file is copied to the host from the inner namespace (if that's what you meant to do here)

Reply:
This was symlink in host namespace, but since we don't need it, this line can be removed

Reply:
kk